### PR TITLE
feat: `SuperchainETHWrapper` contract

### PR DESCRIPTION
**Pull request by tremarkley**
Closes https://github.com/ethereum-optimism/supersim/issues/135

This contract facilitates sending ETH across chains within the Superchain by wrapping ETH into `SuperchainWETH`, relaying the `SuperchainWETH` to the destination chain using `L2ToL2CrossDomainMessenger`, and then unwrapping it back to ETH and sending it to the specified recipient on the destination chain. This is not meant to be used in production and should only be used in testnet or for prototypes.

### CODE REVIEW

**Comment by tremarkley** [comment]
* **#167** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/supersim/167?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tremarkley and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by zainbacchus** [comment]
OH LETS GOOOOOO

File: interop/relayer.go [hunk 1]
```diff
@@ -119,16 +117,11 @@ func (r *L2ToL2MessageRelayer) Start(indexer *L2ToL2MessageIndexer, clients map[
 					}
 					// Pad gas by 33%.
 					paddedGas := (gasEstimate / 3) + gasEstimate
-					mu.Lock()
```
Comment (tremarkley) [inline interop/relayer.go:122]: this mutate does not work properly when multiple messages are sent in the same transaction. simplest way to fix this is to remove the lock and move the transactor within the case block

**Comment by tremarkley** [comment]
Once https://github.com/ethereum-optimism/optimism/pull/12514 merges, then I will update this to include latest changes to `SuperchainWETH`

File: contracts/src/SuperchainETHWrapper.sol [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * @dev The contract integrates with the SuperchainWETH contract for wrapping and unwrapping ETH,
+ *      and uses the L2ToL2CrossDomainMessenger for relaying the wrapped ETH between chains.
+ */
+contract SuperchainETHWrapper {
+    /**
+     * @dev Emitted when ETH is received by the contract.
+     * @param from The address that sent ETH.
+     * @param value The amount of ETH received.
+     */
+    event LogReceived(address from, uint256 value);
```
Comment (nitaliano) [inline contracts/src/SuperchainETHWrapper.sol:39]: might be overkill but having the `from` address indexed might be helpful for debugging

**Review by nitaliano — APPROVED** [review]

File: contracts/src/SuperchainETHWrapper.sol (hunk 2 above)
Comment (hamdiallam) [reply contracts/src/SuperchainETHWrapper.sol:39]: rename to `EthReceived()`?

**Review by hamdiallam — APPROVED** [review]

File: contracts/src/SuperchainETHWrapper.sol (hunk 2 above)
Comment (tremarkley) [reply contracts/src/SuperchainETHWrapper.sol:39]: done

**Pull request merged**